src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotMetadataStep.java

```diff
@@ -45,14 +46,23 @@ public CreateSnapshotMetadataStep(
   public StepResult doStep(FlightContext context) {
     try {
       FlightMap workingMap = context.getWorkingMap();
-      // fill in the ideas that we made in previous steps
+      // fill in the ids that we made in previous steps
       UUID projectResourceId =
           workingMap.get(SnapshotWorkingMapKeys.PROJECT_RESOURCE_ID, UUID.class);
       Snapshot snapshot =
           snapshotService
               .makeSnapshotFromSnapshotRequest(snapshotReq)
               .id(snapshotId)
               .projectResourceId(projectResourceId);
+      if (snapshotReq
+          .getContents()
+          .get(0)
+          .getMode()
+          .equals(SnapshotRequestContentsModel.ModeEnum.BYREQUESTID)) {
```

Comment:
When comparing enums, I think it's better to use `==` instead of `equals()`, as enum instances are final statics.
```suggestion
          == SnapshotRequestContentsModel.ModeEnum.BYREQUESTID) {
```